test/functional/interface_ipc.py

```diff
@@ -0,0 +1,166 @@
+            assert blockref.result.height > 100
+            # Mine a block.
+            wait = mining.result.waitTipChanged(ctx, blockref.result.hash, 1000.0)
+            self.generate(self.nodes[0], 1)
+            newblockref = await wait
+            assert_equal(len(newblockref.result.hash), 32)
+            assert_equal(newblockref.result.height, blockref.result.height + 1)
+            # Wait for timeout (1000 milliseconds)
+            wait = mining.result.waitTipChanged(ctx, newblockref.result.hash, 1000.0)
+            oldblockref = await wait
+            assert_equal(len(newblockref.result.hash), 32)
+            assert_equal(oldblockref.result.hash, newblockref.result.hash)
+            assert_equal(oldblockref.result.height, newblockref.result.height)
+
+            # Create a template.
+            opts = self.capnp_modules['mining'].BlockCreateOptions()
+            opts.useMempool = True
+            opts.blockReservedWeight = 4000
+            opts.coinbaseOutputMaxAdditionalSigops = 0
+            template = mining.result.createNewBlock(opts)
+            # Test some inspectors of template.
+            header = await template.result.getBlockHeader(ctx)
+            assert_equal(len(header.result), 80)
+            block = await self.parse_and_deserialize_block(template, ctx)
+            assert_equal(ser_uint256(block.hashPrevBlock), newblockref.result.hash)
+            assert len(block.vtx) >= 1
+            txfees = await template.result.getTxFees(ctx)
+            assert_equal(len(txfees.result), 0)
+            txsigops = await template.result.getTxSigops(ctx)
+            assert_equal(len(txsigops.result), 0)
+            coinbase_data = BytesIO((await template.result.getCoinbaseTx(ctx)).result)
+            coinbase = CTransaction()
+            coinbase.deserialize(coinbase_data)
+            assert_equal(coinbase.vin[0].prevout.hash, 0)
+            # Wait for a new template.
+            waitoptions = self.capnp_modules['mining'].BlockWaitOptions()
+            waitoptions.timeout = 1000.0
+            waitnext = template.result.waitNext(ctx, waitoptions)
+            self.generate(self.nodes[0], 1)
+            template2 = await waitnext
+            block2 = await self.parse_and_deserialize_block(template2, ctx)
+            assert_equal(len(block2.vtx), 1)
+            template3 = None
+            last_template = template2
+            if self.uses_wallet:
+                # Wait for another, get one after increase in fees in the mempool.
+                waitnext = template2.result.waitNext(ctx, waitoptions)
+                self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 1, fee_rate=10)
+                template3 = await waitnext
+                block3 = await self.parse_and_deserialize_block(template3, ctx)
+                assert_equal(len(block3.vtx), 2)
+                last_template = template3
+            # Wait for another, but timeout.
```

Comment:
088dc2c486af0ad6803d919d8114ab29d3cd652f: nit if you have to retouch: these comments are useful as `self.log.debug()`